### PR TITLE
Added Support for user specificed CA file and url encoding of JMX pat…

### PR DESCRIPTION
**Pull request by ninjasftw**
I'm using this plugin to monitor a number of tomcat instances.  
There were a couple of issues for my use case that I've fixes and would like to push back into your project.  
*  All our jmx interfaces are ssl protected by custom signed certificates so I needed a way to provide a CA location 
*  I've added url encoding to the JMX urls so that beans such as java.lang:type=GarbageCollector,name=PS Scavenge work natively without needing to encode the whitespace

### CODE REVIEW

**Comment by radoondas** [comment]
Hi @ninjasftw !
Thanks a lot for your PR. I appreciate it much and I'm glad you found use for it. :)
I'll review it soon.

File: main.go [hunk 1]
```diff
@@ -5,7 +5,7 @@ import (
 
 	"github.com/elastic/beats/libbeat/beat"
 
-	"github.com/radoondas/jmxproxybeat/beater"
+	"github.com/ninjasftw/jmxproxybeat/beater"
 )
```
Comment (radoondas) [inline main.go]: This import should stay from original repo. Could you fix it?

File: beater/jmxproxybeat.go [hunk 2]
```diff
@@ -9,7 +9,7 @@ import (
 	"github.com/elastic/beats/libbeat/cfgfile"
 	"github.com/elastic/beats/libbeat/logp"
 	"github.com/elastic/beats/libbeat/publisher"
-	"github.com/radoondas/jmxproxybeat/config"
+	"github.com/ninjasftw/jmxproxybeat/config"
```
Comment (radoondas) [inline beater/jmxproxybeat.go]: This import should stay from original repo. Could you fix it?

**Comment by radoondas** [comment]
Thanks for URL encoding update! It was already on my list.

**Comment by ninjasftw** [comment]
Hi @radoondas 
I've changed the package imports back to your tree.  Need to remember to change that as I can't build it when linking to yours.

**Comment by radoondas** [comment]
Hi @ninjasftw Thanks! Will merge it now as is. Looks very good. 
Just for my curiosity, what is number of Tomcats you are querying?

**Comment by ninjasftw** [comment]
Hi @radoondas Tomcat only makes up a fairly small portion of my estate, ~30 or so.  I do however have lots of other middleware products that i'm looking to move over to Elastic monitoring/reporting soon.

**Comment by radoondas** [comment]
@ninjasftw Thanks. It makes me proud that the tool is used. I am working on another similar beat for JBoss/Wildfly. Stay tuned. :)

**Comment by ninjasftw** [comment]
Jboss one could be handy for me.  I'm currently using a fork of this to create one for Websphere Liberty Profile.  Good luck :)

**Comment by radoondas** [comment]
@ninjasftw Good luck with your beat :)

**Comment by radoondas** [comment]
@ninjasftw Since you are working on your beat based on this one you should know that I have pushed new branch which will be compatible with libbeat version 5  (Beats 5). Please have a look and I hope it will help you as well.  
Branch: beats_v5 
Once I'm sure all is fine, I'll merge it with master. Most probably around the time of Beats5_beta
Have fun.

**Pull request merged**